GPS LNA for Rocket

noritsuna opened the pull request:
I am currently working with Kagoshima University in Japan on the joint development of a GPS system for rockets. Therefore, we designed three types of LNAs to acquire various types of data with the aim of making the GPS system a one-chip system in the future. The first design was to design an LNA based on an NPN transistor. Second design, because the first circuit is sensitive to inductors, an LNA with an external inductor was designed. Third design, the NPN transistor in the second circuit was replaced by a CMOS, and an LNA was designed with the same characteristics. These measurements will be used in the design of future one-chip GPS systems for rockets.

noritsuna committed:
1st version.

1st version.

noritsuna commented:
GDS for submission is /GPS_LNA/design_data/gds/gps_lns_fill.gds 

In the DRC, we are seeing three AFil.g2 errors, but we do not think there is a problem.
See /GPS_LNA/doc/source/validation.rst for details. 
Please let me know if there is a way to address this.

mole99 commented:
You could give the magic filler insertion script a try, for my design it worked great :)

Here is the PR: https://github.com/IHP-GmbH/IHP-Open-PDK/pull/409
You need the latest version of magic to run it.

noritsuna commented:
> You could give the magic filler insertion script a try, for my design it worked great :)
> 
> Here is the PR: [IHP-GmbH/IHP-Open-PDK#409](https://github.com/IHP-GmbH/IHP-Open-PDK/pull/409) You need the latest version of magic to run it.

The Magic has a read error in my GDS...

![image](https://github.com/user-attachments/assets/7b5c6be0-9078-4443-9531-452811310797)

KrzysztofHerman commented:
@noritsuna your design fails on this step. Please fix it
```
ERROR: RuntimeError: 'source': The layout has multiple top cells in Layout::top_cell in Executable::execute
  /home/runner/work/TO_Apr2025/TO_Apr2025/drc/drc.lydrc:34:in `execute'
  :/built-in-macros/drc_interpreters.lym:31:in `instance_eval'
  :/built-in-macros/drc_interpreters.lym:31:in `execute'
```

noritsuna commented:
> @noritsuna your design fails on this step. Please fix it
> 
> ```
> ERROR: RuntimeError: 'source': The layout has multiple top cells in Layout::top_cell in Executable::execute
>   /home/runner/work/TO_Apr2025/TO_Apr2025/drc/drc.lydrc:34:in `execute'
>   :/built-in-macros/drc_interpreters.lym:31:in `instance_eval'
>   :/built-in-macros/drc_interpreters.lym:31:in `execute'
> ```

GDS for submission is /GPS_LNA/design_data/gds/gps_lns_fill.gds
Please drc check to this GDSD file only.
![image](https://github.com/user-attachments/assets/c985a61c-b419-4ce4-8d2d-9b7f2844fa52)

KrzysztofHerman commented:
Dear Contributor, 
since we are closing the submission please change the final GDS name adding the following prefix to the file 
`FMD_QNC_10_`

noritsuna committed:
Add prefix  "FMD_QNC_10_" to final GDS .

Add prefix  "FMD_QNC_10_" to final GDS .

noritsuna commented:
> Dear Contributor, since we are closing the submission please change the final GDS name adding the following prefix to the file `FMD_QNC_10_`

OK. I changed my final GDS file name.

mole99 commented:
> The Magic has a read error in my GDS...

Sorry about that, I added an issue here: https://github.com/IHP-GmbH/IHP-Open-PDK/issues/408

It could be that your three AFil.g2 errors are false positives as they are near the boundary. See: https://github.com/IHP-GmbH/IHP-Open-PDK/issues/404

noritsuna reviewed (commented):
Add prefix "FMD_QNC_10_" to final GDS .

noritsuna commented:
> > The Magic has a read error in my GDS...
> 
> Sorry about that, I added an issue here: [IHP-GmbH/IHP-Open-PDK#408](https://github.com/IHP-GmbH/IHP-Open-PDK/issues/408)
> 
> It could be that your three AFil.g2 errors are false positives as they are near the boundary. See: [IHP-GmbH/IHP-Open-PDK#404](https://github.com/IHP-GmbH/IHP-Open-PDK/issues/404)

Thx!

noritsuna committed:
Delete unused cells.

Delete unused cells.

noritsuna commented:
@KrzysztofHerman 

There are three DRC errors, which I think are bugs in the DRC script.

https://github.com/noritsuna/GPS_LNA/blob/main/GPS_LNA/doc/source/validation.rst
https://github.com/IHP-GmbH/TO_Apr2025/pull/10#issuecomment-2783170616

olisnr commented:
why You dont make some active under the pads? i think this should be legal:

![fill](https://github.com/user-attachments/assets/7bbe463e-4273-46cd-a436-9cfd4e346e9b)

may be it changes the capacity of the pad a bit...

noritsuna committed:
Add activ under the pads.

Add activ under the pads.

noritsuna commented:
> why You dont make some active under the pads? i think this should be legal:
> 
> may be it changes the capacity of the pad a bit...

@olisnr 
Thank you very much.
I inserted Activ under Pad and the DRC error is gone.

![image](https://github.com/user-attachments/assets/cf4b4f55-c58d-477f-a122-4fe0268a63e8)
![image](https://github.com/user-attachments/assets/bfd0cb3e-a68b-4c17-bc57-43d36c33c3d1)

KrzysztofHerman commented:
@noritsuna 

I have verified your design using internal minimal DRC and it is clean and almost ready to be accepted. I would like to kindly ask you for the following modifications:

1) Due to internal issues we have to ask you to open the file in klayout and save it again using `File->Save As..` and provide the following options: 

![image](https://github.com/user-attachments/assets/9c03da59-d0fa-4c82-b2a2-366f080b93f8)

2) Please provide us also some contact data and a shipping address. 

3) Please download the Hiring Agreement from [here](https://ihp-open-ip.readthedocs.io/en/latest/_downloads/c82c58426246cc84f3bbce282975b255/samples.rtf) and sign it. If you are with an university please ask for signature some representative: your advisor, group leader, departments head.  

4) The data mentioned in the points 2-3 you can send to the following [email](openpdk@ihp-microelectronics.com). Please mention this tape out name `TO_Apr2025` in the email title.

noritsuna committed:
Change save options.

Off "Cell context Store PCell and library context information(format specific)"
On "eliminate zero-length paths(convert to BOUNDARY)"

noritsuna commented:
> @KrzysztofHerman 

I changed the GDS options and committed the GDS.
And  I also emailed The Hiring Agreement.

KrzysztofHerman commented:
@noritsuna 
Thank you!

KrzysztofHerman commented:
@noritsuna 

Your design is minimum DRC clean and ready for production.

noritsuna commented:
> Your design is minimum DRC clean and ready for production.

@KrzysztofHerman 
Is there any action I have to do?

KrzysztofHerman commented:
@noritsuna 

Right now just pay attention to the notifications. Tomorrow I should have a final confirmation about the submission. If there will be no notification it means that all is ok.

KrzysztofHerman commented:
@noritsuna 

LBE Mismatch! LBE layer is used. 

Please do not use LBE layer. Remove and resubmit,

noritsuna committed:
Delete BLE layer.

Delete BLE layer.

noritsuna commented:
> @noritsuna
> 
> LBE Mismatch! LBE layer is used.
> 
> Please do not use LBE layer. Remove and resubmit,

@KrzysztofHerman 
I deleted BLE layters.

KrzysztofHerman commented:
@noritsuna 

Congratulations!
The design was accepted for production:
```
File name: FMD_QNC_10_gps_lns_fill_1.gds Top cell name: GPS_LNA
Testfield: T586
Check Summe: 04497e12728450e24faf98fa9d5daa06
Sent: 11-04-2025, 17:15:03
registered size: (1.4 mm X 1.4 mm )
calculated size: (1.35 mm X 1.35 mm))
ID (use this as reference for questions): 54586
```

Please do not change the GDS files anymore.
I will merge this PR to the main branch
